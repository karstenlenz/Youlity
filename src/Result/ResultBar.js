import React from 'react'
import styled, { keyframes } from 'styled-components/macro'

export default function ResultBar({ percentage, index }) {
  return (
    <Outline>
      <ColorBar percentage={percentage} index={index}>
        <Text>{percentage}% Ãœbereinstimmung</Text>
      </ColorBar>
    </Outline>
  )
}

const Text = styled.p`
  margin: 0;
  padding: 8px 15px;
`

const Outline = styled.div`
  border: 1px solid var(--dark-grey);
  border-radius: 5px;
  width: 100%;
`

const expandWidth = (width) => keyframes`
from {
      width: 0;
    }
    to {
      width: ${width}%;
    }
`

const ColorBar = styled.div`
  animation: ${(props) => expandWidth(props.percentage)} 0.8s
    ${(props) => 0.4 + 0.2 * props.index}s forwards ease-in-out;
  white-space: nowrap;
  overflow: visible;
  border-radius: 5px;
  height: 40px;
  width: 0;
  background-color: ${(props) =>
    props.percentage > 50
      ? 'rgba(0,197,170,' + (props.percentage / 50 - 1) + ')'
      : 'rgba(104,59,137,' + props.percentage / 100 + ')'};
`
