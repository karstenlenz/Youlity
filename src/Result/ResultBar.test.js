import { render } from '@testing-library/react'
import React from 'react'
import ResultBar from './ResultBar'

describe('The ResultBar component', () => {
  it('renders correctly', () => {
    const { container } = render(<ResultBar percentage={50} />)
    expect(container).toMatchSnapshot()
  })
})
