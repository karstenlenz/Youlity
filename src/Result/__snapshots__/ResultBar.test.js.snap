// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ResultBar component renders correctly 1`] = `
<div>
  <div
    class="ResultBar__Outline-sc-479x3f-1 echTYD"
  >
    <div
      class="ResultBar__ColorBar-sc-479x3f-2 yIAeb"
    >
      <p
        class="ResultBar__Text-sc-479x3f-0 bCfkyT"
      >
        wenig ausgepr√§gt
      </p>
    </div>
  </div>
</div>
`;
